GitUI/CommandsDialogs/RevisionDiff.cs

```diff
@@ -273,54 +278,43 @@ private void diffShowInFileTreeToolStripMenuItem_Click(object sender, EventArgs
 
         private void DiffContextMenu_Opening(object sender, CancelEventArgs e)
         {
-            bool artificialRevSelected;
-
             IList<GitRevision> selectedRevisions = _revisionGrid.GetSelectedRevisions();
 
-            if (selectedRevisions.Count == 0)
-                artificialRevSelected = false;
-            else
-                artificialRevSelected = selectedRevisions[0].IsArtificial();
-            if (selectedRevisions.Count > 1)
-                artificialRevSelected = artificialRevSelected || selectedRevisions[selectedRevisions.Count - 1].IsArtificial();
-
             //Many options have no meaning for artificial commits or submodules
             //Hide the obviously no action options when single selected, handle them in actions if multi select
 
             // disable items that need exactly one selected item
             bool isExactlyOneItemSelected = DiffFiles.SelectedItems.Count() == 1;
-            var isCombinedDiff = isExactlyOneItemSelected &&
-                DiffFiles.CombinedDiff.Text == DiffFiles.SelectedItemParent;
+            var isCombinedDiff = isExactlyOneItemSelected && DiffFiles.CombinedDiff.Text == DiffFiles.SelectedItemParent;
```

Comment:
I consider this to leave too much logic in the GUI layer. This will not separate the layers, just involve two files instead of one.